fortitude/src/rule_selector.rs

```diff
@@ -129,13 +152,17 @@ pub enum ParseError {
 }
 
 impl RuleSelector {
-    pub fn prefix_and_code(&self) -> (&'static str, &'static str) {
+    pub fn prefix_and_code(&self) -> (String, String) {
```

Comment:
Not sure I understand the reason for this change?

Reply:
The prefixes and codes it returns are macro-generated via `map_codes` and the `RuleNamespace` trait, so they're defined at compile time. To support the deprecated categories, we need to instead return whatever string the user provided.

There's probably a better way to do it by diving into the macros, but I thought that might be overkill when we really just want to redirect everything.